plugins/chat/plugin.rb

```diff
@@ -577,14 +615,21 @@ def self.onebox_template
     end
   end
 
+  on(:user_removed_from_group) do |user, group|
+    Jobs.enqueue(:auto_remove_membership_handle_user_removed_from_group, user_id: user.id)
+  end
+
   on(:category_updated) do |category|
     # TODO(roman): remove early return after 2.9 release.
     # There's a bug on core where this event is triggered with an `#update` result (true/false)
-    return if !category.is_a?(Category)
-    category_channel = ChatChannel.find_by(auto_join_users: true, chatable: category)
+    if category.is_a?(Category) && category_channel = ChatChannel.find_by(chatable: category)
```

Comment:
Had to keep this as a normal `if` statement, early returns caused this error to occur in specs:

```
 25) Chat::Api::ChatChannelThreadsController show when thread exists when user cannot access the channel returns 403
      Failure/Error: DiscourseEvent.on(event_name) { |*args, **kwargs| block.call(*args, **kwargs) if enabled? }

      LocalJumpError:
        unexpected return
      # ./plugins/chat/plugin.rb:627:in `block (2 levels) in activate!'
      # ./lib/plugin/instance.rb:491:in `block in on'
      # ./lib/discourse_event.rb:12:in `block in trigger'
      # ./lib/discourse_event.rb:12:in `trigger'
      # ./spec/support/discourse_event_helper.rb:5:in `trigger'
      # ./app/models/category.rb:947:in `block (2 levels) in <class:Category>'
```

Not sure how the original `return if !category.is_a?(Category)` worked, maybe there was no spec for it or it just never happened.

Reply:
Ah yes, that’s because we’re in a block context. Just replace `return` by `break` and it should work as expected :slightly_smiling_face: